lib/fluent/supervisor.rb

```diff
@@ -134,13 +135,20 @@ def run_rpc_server
 
       @rpc_server.mount_proc('/api/config.gracefulReload') { |req, res|
         $log.debug "fluentd RPC got /api/config.gracefulReload request"
-        if Fluent.windows?
-          supervisor_sigusr2_handler
+        if @reload_after_timestamp && @reload_after_timestamp > Time.now
+          $log.debug("accept grcefulReload after: #{@reload_after_timestamp} remaining: #{(@reload_after_timestamp - Time.now).to_i} seconds")
+          [422, nil, {'message': 'failed to reload config'}]
         else
-          Process.kill :USR2, $$
-        end
+          @reload_after_timestamp = Time.now + 60
```

Comment:
I'm not sure 60 seconds is suitable or not.
(IMHO I feel it's too long)

Reply:
I think that this value should be configurable such as `blocking_reload_interval` or something else in system config.
If it is configurable, we can specify arbitrary value for blocking reload. 
I feel that the half of its value is enough but no evidence yet.

Reply:
I agree with it. it should be customizable to this kind of parameter.